packages/code-du-travail-frontend/src/common/CompanySuggester.js

```diff
@@ -0,0 +1,51 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Suggester } from "./Suggester";
+import {
+  renderSuggestionsContainer,
+  Suggestion,
+  suggesterTheme
+} from "./IdccSuggester";
+
+export class CompanySuggester extends React.Component {
+  static propTypes = {
+    onSearch: PropTypes.func.isRequired,
+    onSelect: PropTypes.func.isRequired
+  };
+  onSelect = value => {
+    this.props.onSelect(value);
+  };
+  render() {
+    return (
+      <Suggester
+        onSearch={this.props.onSearch}
+        onSelect={this.onSelect}
+        renderSuggestion={renderSuggestion}
+        renderSuggestionsContainer={renderSuggestionsContainer}
+        placeholder="Numéro de SIRET à 14 chiffres"
+        theme={suggesterTheme}
+        // inserts spaces to have a "000 000 000 00000" format and limits to 14 chars
+        reformatEnteredValue={v =>
+          v
+            .toString()
+            .replace(/ /g, "")
+            .substring(0, 14)
+            .replace(/(\d{1,3})?(\d{1,3})?(\d{1,3})?(\d{1,5})?/, "$1 $2 $3 $4")
```

Comment:
This is (hard)core business transformation ! 
should be isolated and (battle) tested in unit tests

Reply:
tu peux exporter la fonction 
```
export reformatCompagnyNum = v => 
  v
    .replace(/ /g, "")
    .substring(0, 14)
    .replace(/(\d{1,3})?(\d{1,3})?(\d{1,3})?(\d{1,5})?/, "$1 $2 $3 $4")
```

ca permettra effectivement de tester la fonction

Reply:
ok!